### ISSUE TITLE
Can't use required/ignore words after removeWords feature

### ISSUE THREAD

**Issue by deusxanime** [SiCKRAGETV/sickrage-issues#1294]
I've just noticed on my SickRage install that my SD downloads, which I require keywords 'eztv' or 'ettv', are no longer working, similar to what is detailed [here](https://github.com/SiCKRAGETV/sickrage-issues/issues/246). Looks like it started sometime in the past week or so as up until recently it had been working fine. I can look on KAT and others and see the EZTV and ETTV releases there just fine but it seems to completely skip over them now when searching.

SR Version: BRANCH: (master) / COMMIT: (281409d366b228a455879e02b2d55d3967db29e4) 
Python Version: 2.7.9 (default, Dec 10 2014, 12:24:55) [MSC v.1500 32 bit (Intel)]
Running on Windows 7 32-bit

Example series this happens with is "Grimm", though it is not grabbing any of my SD series as they all are setup the same way with the same Required Words:
```
Quality:	Initial: SD TV
Originally Airs:	Friday 8:00 PM on NBC
Show Status:	Continuing
Default EP Status:	Archived
Location:	V:\tv\Grimm
Scene Name:	Grimm
Required Words:	ettv,eztv
Ignored Words:	XviD
Size:	24.78 GB
```

Log of a self-initiated [manual search](https://gist.github.com/deusxanime/bdfc39313643f1a767a2).

Picture showing current search on KAT:
![image](https://cloud.githubusercontent.com/assets/11891866/7193596/abd25878-e468-11e4-9bd5-f1940364ec86.png)

I've run the updater and it says I am at the lastest version as of yesterday and have tried restarting SickRage also.

**Comment by fernandog** [comment]
@tagforce @abeloin we should remove only after we snatch and not before. Otherwise user wont be able to use required words with in this case

https://github.com/SiCKRAGETV/SickRage/pull/1703

**Comment by TagForce** [comment]
@abeloin @fernandog Well, this has come back to bite us in the bee-hind...
Revert all those changes. We should go about this differently methinks.
Either add an (optional) field in the regex (to catch \[\.+\] at the end of the releasenames so that releasegroup doesn't get polluted with the provider tags, or simply remove them from the releasegroup in PP.

**Comment by deusxanime** [comment]
Since this seems to have to do with reading tags enclosed in brackets [ ], could this be affecting anime also? Or is that handled differently? I started trying to utilize SR for my anime as well now for the Spring 2015 season (started a couple weeks ago) and have been having issues trying to get it to grab some releases. It seems sparodic though, sometimes it does seem to detect them, sometimes not. Especially wasn't working with [FFF] releases at one point, which about half the series I'm getting are done by them, though it has grabbed a few of them now I believe.

Anyway just throwing that out there, though I know anime is a different beast with different naming scene/fansub group schemes, each season having different names, absolute ordering, and other fun stuff.

**Comment by fernandog** [comment]
@deusxanime don't affect anime

**Comment by deusxanime** [comment]
@fernandog Ok thanks for the info. I'll keep fiddling with that and if it becomes a problem I'll open a different issue. Right now still just playing around with the anime support.

**Comment by T-2000** [comment]
Just encountered something similar, searching and post processing appears to be not working for this particular series, post processing log below, don't exactly know what's going on, i've tried using the anime check box, adding alternate scene names and white listing fansub groups.
After changing the settings it still didn't work so i tried removing the show from SR and re-adding it after re-scraping the meta info using Ember Media Manager.

>Processing folder /volume1/dstation
TV_DOWNLOAD_DIR: /volume1/dstation
PostProcessing Path: /volume1/dstation
PostProcessing Dirs: []
PostProcessing Files: ['[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv']
PostProcessing VideoFiles: ['[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv']
PostProcessing RarContent: []
PostProcessing VideoInRar: []
Processing /volume1/dstation/[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv (None)
Retrieving episode object for 1x2
Snatch history had a quality in it, using that: 1080p HD TV
Is ep a priority download: False
File /volume1/videos/anime/airing/Nisekoi/S01/Nisekoi - 1x02 - Encounter.mkv is smaller than /volume1/dstation/[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv
Deleting file /volume1/videos/anime/airing/Nisekoi/S01/Nisekoi - 1x02 - Encounter.mkv
Deleting file /volume1/videos/anime/airing/Nisekoi/S01/Nisekoi - 1x02 - Encounter-thumb.jpg
Deleting file /volume1/videos/anime/airing/Nisekoi/S01/Nisekoi - 1x02 - Encounter.nfo
Found release name [HorribleSubs] Nisekoi S2 - 02 [1080p]
Destination folder for this episode: /volume1/videos/anime/airing/Nisekoi/S01
Adding the file to the anidb mylist
Moving file from /volume1/dstation/[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv to /volume1/videos/anime/airing/Nisekoi/S01/Nisekoi - 1x02 - Encounter.mkv
Processing succeeded for /volume1/dstation/[HorribleSubs] Nisekoi S2 - 02 [1080p].mkv
Successfully processed

**Comment by abeloin** [comment]
@fernandog @TagForce Redoing the regex should be best long term solution

**Comment by fernandog** [comment]
@miigotu can you help with this one?

**Comment by miigotu** [comment]
Those results he took a pic of on KAT aren't SDTV. ^^^

**Comment by deusxanime** [comment]
That is the way the scene has labeled them forever though. They are HDTV rips (sourced from HD broadcast), but encoded in SD. In my screenshot, those are the top 2. Otherwise they'd have the "720p" label on them, like the other 2.

**Comment by miigotu** [comment]
@fernandog is correct.

This is kind of a PITA to fix cleanly lol. Why do we need to strip rls groups from the title anyways?

**Comment by fernandog** [comment]
@miigotu PP issues. SR considers -DIMMENSION and -DIMMENSION[rartv] different release groups

**Comment by miigotu** [comment]
Ok, but rather than stripping them in in a handful of providers as it is now, and not having a complete title for the req/ign words, why aren't they stripped in the PP instead? This is a rather involved fix, above my pay scale I think lolz

**Comment by fernandog** [comment]
@abeloin suggested to redo the regex but don't know how that fill fix it? Maybe split release group in two groups ? release name and the tracker name, in this case [rartv]. But how to make this tracker name group optional?

**Comment by miigotu** [comment]
It has nothing to do with the regex, thats fine. It's that its simply stripped off of the title and overwrites the original title. The removed words are no longer stored in the object to check against.

**Comment by miigotu** [comment]
I tried a few things that didnt work so nice.  But, I have another idea, let me try it out.

**Comment by miigotu** [comment]
Eh, it's over my head, or I'm too lazy to change every provider.

**Comment by fernandog** [comment]
what's your idea?

**Comment by miigotu** [comment]
My ideas didn't work. I added a removedWords to the generic provider, and appended words that were stripped, but that way it would be appended with each result and just be useless for individual results.

Really the only way to make it work as it sits is to modify every provider, and pass a new argument down through and back up through 3-4 classes. I think that is a dirty way to do it, and it seems a little dirty already.

A massive rework of code is just not my cup of tea right now, and I'd be worried I'd break something.

**Comment by fernandog** [comment]
Why not update regex and split releasename in too? Create a new optional group "tracker name"

**Comment by deusxanime** [comment]
I've gotten around this problem by changing my Required Words to "x264" which seems to be grabbing the releases I want. Previously it was grabbing .avi and web-dl files which I didn't want, which was why I stuck in the eztv,ettv Required Words. Anyway just having x264 seems to be working. 

This is still an interesting problem, and I'd prefer to have the eztv or ettv releases as I know they are consistently quality/size that I want, so it would be nice to have those Required Words work. Up to you how much you want to continue to pursue this for everyone's sake and not just my own though.

**Comment by fernandog** [comment]
@TagForce what do you think about this: https://github.com/SiCKRAGETV/sickrage-issues/issues/1294#issuecomment-95306277

will it have huge impact in PP?

**Comment by TagForce** [comment]
@fernandog Yes, kind of... We will have to go through every possible codepath for PP, and add the extra field to each function that determines destination filenames. Luckily that's not that many.
The provider field would be mostly unused (and can be ignored, but we need to check regardless).
The only real place where we'd need to add code for it is this function:
https://github.com/SiCKRAGETV/SickRage/blob/master/sickbeard/tv.py#L2252

We'd also need to include the field in the naming patterns, so people can add it to the filename if they want. Probably something like %PT and %pt for upper and lower case "provider tag".

All in all, quite some work.

**Comment by miigotu** [comment]
I have this fixed, pushing a PR in about 5 mins.

**Comment by fernandog** [comment]
Thanks!
